common/src/main/java/com/revenuecat/purchases/common/verification/SignatureVerificationMode.kt

```diff
@@ -19,8 +19,18 @@ sealed class SignatureVerificationMode {
         }
     }
     object Disabled : SignatureVerificationMode()
-    data class Informational(val signatureVerifier: SignatureVerifier) : SignatureVerificationMode()
-    data class Enforced(val signatureVerifier: SignatureVerifier) : SignatureVerificationMode()
+    data class Informational(
+        val signatureVerifier: SignatureVerifier,
+        override val intermediateSignatureHelper: IntermediateSignatureHelper = IntermediateSignatureHelper(
+            signatureVerifier,
+        ),
+    ) : SignatureVerificationMode()
+    data class Enforced(
+        val signatureVerifier: SignatureVerifier,
+        override val intermediateSignatureHelper: IntermediateSignatureHelper = IntermediateSignatureHelper(
+            signatureVerifier,
+        ),
```

Comment:
I think it's slightly weird that this has both of these types. Why would `IntermediateSignatureHelper` not be inside `SignatureVerifier`?

Reply:
Oh I see `SigningManager` is the type on top. But it seems like it only uses `IntermediateSignatureHelper` though?

Reply:
I removed the `signatureVerifier` field from the enum here: https://github.com/RevenueCat/purchases-android/pull/1111/commits/9efce80dc41f39917ec58c99e10a4664652aed5a. Lmk what you think!